src/createPackage/models/createPackageManager.ts

```diff
@@ -338,6 +343,81 @@ export class CreatePackageManager {
     await fsPromise.writeFile(metadataFilePath, recordMetadata);
   }
 
+  public async createExportJsonMetadata(job: JobExportResponse): Promise<void> {
+    this.logger.info(`Creating metadata.json file for gpkg in path "${job.parameters.relativeDirectoryPath}" for jobId ${job.id}`);
+    const record = await this.rasterCatalogManager.findLayer(job.internalId as string);
+    const featuresRecords = parseFeatureCollection(job.parameters.roi);
+
+    const maxResolutionDeg = Math.max(
+      record.metadata.maxResolutionDeg as number,
+      Math.min(...featuresRecords.map((records) => records.targetResolutionDeg))
+    );
+    const maxResolutionMeter = Math.max(
+      record.metadata.maxResolutionMeter as number,
+      Math.min(...featuresRecords.map((records) => records.targetResolutionMeter))
+    );
+
+    const metadataFileName = job.parameters.fileNamesTemplates.metadataURI;
+    const directoryName = job.parameters.relativeDirectoryPath;
+    const metadataFullPath = concatFsPaths(this.gpkgsLocation, directoryName, metadataFileName);
+    let combinedFootprint = undefined;
+    try {
+      const intersectedFeatures = this.featuresFootprintIntersects(
+        job.parameters.roi.features as Feature<Polygon | MultiPolygon>[],
+        record.metadata.footprint as Polygon | MultiPolygon
+      );
+      const fc: FeatureCollection<Polygon | MultiPolygon> = featureCollection(intersectedFeatures);
+      combinedFootprint = featureCombine(fc).features[0].geometry as unknown as MultiPolygon;
+    } catch (error) {
+      this.logger.error(error, `Failed to match features intersection with footprint`);
+    }
+    record.metadata.footprint = combinedFootprint ? combinedFootprint : record.metadata.footprint;
+    record.metadata.maxResolutionDeg = maxResolutionDeg;
+    record.metadata.maxResolutionMeter = maxResolutionMeter;
+
+    const layerPolygonPartFeatures: Feature[] = [];
```

Comment:
refactor to getExportedPackageLayerPolygonParts method

Reply:
done